nemo/core/config/set_config.py

```diff
@@ -24,15 +24,22 @@
 
 
 def hydra_runner(
-    config_path: Optional[str] = None, config_name: Optional[str] = None
+    config_path: Optional[str] = None, config_name: Optional[str] = None, schema: Optional[Any] = None
 ) -> Callable[[TaskFunction], Any]:
     """
     Decorator used for passing the Config paths to main function.
+    Optionally registers a schema used for validation/providing default values.
 
     Args:
-        config_path: path to the directory where the config exists.
-        config_name: name of the config file.
+        config_path: Path to the directory where the config exists.
+        config_name: Name of the config file.
+        schema: Structured config  type representing the schema used for validation/providing default values.
     """
+    if schema is not None:
+        # Create config store.
+        cs = ConfigStore.instance()
+        # Register the configuration as a node under a given name.
+        cs.store(name=config_name, node=schema)
```

Comment:
Config name can be a yaml file or just a file name. Maybe check and remove the .yaml from here config name when setting it as the name in the store.

Reply:
Also, as above, why is this registration necessary? OmegaConf has structured() no?

Reply:
Hmmmm, I wanted to rely on hydra as much as possible - and IMO it was the simplest solution.

But if you want to reimplement it in a different way, please go ahead. I just want to have option to inject the schema.

Reply:
I'm just saying clean up the name. name = comfig_name.replace(".yaml", "") will suffice

Reply:
ok